### PR TITLE
Remove hammer and use native touch events

### PR DESCRIPTION
**Pull request by rogatty**
https://wikia-inc.atlassian.net/browse/HG-502

Same as https://github.com/Wikia/mercury/pull/487 but with `dev` branch as target.

### CODE REVIEW

File: public/scripts/main/views/MediaLightboxView.ts [hunk 1]
```diff
@@ -223,60 +216,153 @@ App.MediaLightboxView = App.LightboxView.extend({
...
 	},
 
-	gestures: {
-		swipeLeft: function (): void {
-			if (this.get('isGallery') && !this.get('isZoomed')) {
-				this.nextMedia();
-			}
-		},
+	gesture: {
+		minSwipe: 50,
```
Comment (rogatty) [inline public/scripts/main/views/MediaLightboxView.ts]: This is being set in two places (for now). Should we abstract it now or wait until there are more cases?

File: public/scripts/main/views/MediaLightboxView.ts [hunk 2]
```diff
@@ -223,60 +216,153 @@ App.MediaLightboxView = App.LightboxView.extend({
...
+	 * resets data after pinch and pan
+	 * @param event
+	 * @returns {boolean}
+	 */
+	touchEnd: function (event: {originalEvent: {touches: any[]}}): void {
+		var x = Em.get(event, 'originalEvent.changedTouches.0.clientX');
+
+		//swipe
+		if (this.get('isGallery') && !this.get('isZoomed')) {
+			if (Math.abs(this.get('gesture.startX') - x) > this.gesture.minSwipe) {
```
Comment (rogatty) [inline public/scripts/main/views/MediaLightboxView.ts]: How about abstracting this condition to utils? Then all the settings could be there too.

> @hakubo:
> I am not sure about overloading utils module
> maybe a mixin could help us somehow :)

File: front/scripts/main/views/ArticleView.ts [hunk 3]
```diff
@@ -172,37 +172,43 @@ App.ArticleView = Em.View.extend(App.AdsMixin, {
 		}
 	},
 
-	hammerOptions: {
-		touchAction: 'auto'
+	minSwipe: 50,
```
Comment (kenkouot) [inline front/scripts/main/views/ArticleView.ts]: Are these intentionally just for the article view? Seems like they should be application wide?

Comment (hakubo) [inline front/scripts/main/views/ArticleView.ts]: Yes, or even mixin, but I would test it this way.
And if it works and if we use this in more places, then I would abstract it away.

File: public/scripts/main/views/MediaLightboxView.ts (hunk 1 above)
Comment (hakubo) [inline public/scripts/main/views/MediaLightboxView.ts]: I would wait, I would like to see if this implementation does not degrade UX now.

File: public/scripts/main/views/MediaLightboxView.ts (hunk 2 above)
Comment (hakubo) [inline public/scripts/main/views/MediaLightboxView.ts]: When we will have more places with such interaction, We will create a mixin for that

**Comment by hakubo** [comment]
@tomasznapieralski Could you let me know how testing went? Better then previous interaction? Worse?

**Comment by hakubo** [comment]
It was fun to try it,
But I don't think we should maintain another implementation of gestures
hammer is good enough for us, if wee need something else we can always add it to it

**Pull request closed without merging**